internal/controller/port_mapper.go

```diff
@@ -67,17 +66,29 @@ type PortMapper interface {
 }
 
 type PortMapperImpl struct {
-	l3manager openstack.L3PortManager
+	l3manager L3PortManager
 	services  map[string]model.ServiceModel
 	l3ports   map[string]model.L3Port
 }
 
-func NewPortMapper(l3manager openstack.L3PortManager) PortMapper {
-	return &PortMapperImpl{
+func NewPortMapper(l3manager L3PortManager) PortMapper {
+	portManager := &PortMapperImpl{
 		l3manager: l3manager,
 		services:  make(map[string]model.ServiceModel),
 		l3ports:   make(map[string]model.L3Port),
 	}
+
+	// Load all available ports
+	l3portIDs, err := l3manager.GetAvailablePorts()
+	if err == nil {
+		for _, l3portID := range l3portIDs {
+			portManager.emplaceL3Port(l3portID)
+		}
+	} else {
+		klog.Warningf("port mapper could not load available l3ports: %s", err)
```

Comment:
Should this be fatal? Can we operate safely without the port listing?

Mind that we're running in k8s, so if we bail out here (by returning err and handling that by exiting in main.go), we'd be restarted after a moment and problems would at least be visible.

Reply:
We can operate more or less safely.

The OpenStack port manager can work without it, but #42 would be a small problem again.

The Static port manager always needs the list of available ports in the beginning, but the `GetAvailablePorts` method cannot return an error in this case.

But if the listing of the available ports from OpenStack does not work, other things may not work too. So we can make this fatal.